### PR TITLE
Fix MS_Reachability dealloc

### PR DESCRIPTION
**Pull request by MatkovIvan**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

`Thread 1: EXC_BAD_ACCESS (code=1, address=0x15ef3e157958)” in MS_Reachability class (line 149)`

## Related PRs or issues

#1293

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 ## Version 1.13.2 (Not yet released)
 
-* **[Fix]** Fix a crash if database query failed. 
+* **[Fix]** Fix a crash if database query failed.
+* **[Fix]** Fix a crash on simulator that was introduced in 1.13.1.
```
Comment (jaelim-ms) [inline CHANGELOG.md]: Can you give more details instead of just saying "was introduced in 1.13.1"? Adding a reason of crash would be enough.

**Comment by clpolet** [comment]
Marked as don't merge until we have a better understanding of the repro/condition of the issue.

File: AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m [hunk 2]
```diff
@@ -124,7 +133,7 @@ + (instancetype)reachabilityForInternetConnection {
 #pragma mark - Start and stop notifier
 
 - (void)startNotifier {
-  dispatch_block_t block = ^{
+  RunOnMainThread(^{
```
Comment (achocron) [inline AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m]: Can you just add an explanation comment for why it needs to run on the main thread?

File: AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m [hunk 3]
```diff
@@ -136,32 +145,25 @@ - (void)startNotifier {
...
+  });
 }
 
 - (void)dealloc {
-  [self stopNotifier];
-  if (self.reachabilityRef != NULL) {
-    CFRelease(self.reachabilityRef);
+  __block SCNetworkReachabilityRef reachabilityRef = self.reachabilityRef;
+  if (reachabilityRef != NULL) {
+    RunOnMainThread(^{
```
Comment (achocron) [inline AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m]: here too

File: AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m [hunk 4]
```diff
@@ -54,6 +54,15 @@ static void ReachabilityCallback(SCNetworkReachabilityRef target,
                     object:noteObject];
 }
 
+
```
Comment (jwallra) [inline AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m]: nit: remove extra blank line

File: AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m (hunk 2 above)
Comment (MatkovIvan) [reply AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m]: https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1307/files/218d4e78cd9d3735c753e349c760db3f1a3e1f06#diff-961cca420883c03f21679b93b30e250fR77

The behaviour isn't changed in this PR

**Comment by clpolet** [comment]
We'll need a unit test to reproduce the issue and make sure this won't happen in the future.

File: AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m [hunk 5]
```diff
@@ -124,7 +133,7 @@ + (instancetype)reachabilityForInternetConnection {
 #pragma mark - Start and stop notifier
 
 - (void)startNotifier {
-  dispatch_block_t block = ^{
+  RunOnMainThread(^{
     SCNetworkReachabilityContext context = {0, (__bridge void *)(self), NULL,
                                             NULL, NULL};
     if (SCNetworkReachabilitySetCallback(self.reachabilityRef,
```
Comment (clpolet) [inline AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m]: Also the pattern in using `self` in blocks is to use weak self as we do in other places in the SDK.

Comment (clpolet) [reply AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m]: Although if we use weakself the ref might never be unscheduled but will be freed so you are probably right not to use weakself. Just double check maybe.

Comment (MatkovIvan) [reply AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m]: This approach lives since GA at least. It checked a lot of times

**Pull request merged**